### PR TITLE
[DPE-2636] Mongodb 6 with 6/edge and server built from source

### PR DESCRIPTION
**Pull request by phvalguima**

### CODE REVIEW

**Comment by phvalguima** [comment]
Moving to review as the before last commit passed all tests and the last commit is just a documentation update.

**Review by MiaAltieri — CHANGES_REQUESTED** [review]
Amazing work 🤩 only one change needed, [change CI to `6/edge`](https://github.com/canonical/mongodb-operator/blob/92f35e4bcc07d31c7f64bbaa4515eb9bf39441dc/.github/workflows/release.yaml#L34-L41), the track has already been [approved on charmhub](https://discourse.charmhub.io/t/request-track-6-for-charmed-mongodb-operator/11900/3)

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -31,11 +31,9 @@ jobs:
...
-      #   with:
-      #     credentials: "${{ secrets.CHARMHUB_TOKEN }}"
-      #     github-token: "${{ secrets.GITHUB_TOKEN }}"
-      #     channel: "6/edge"
+      - name: Upload charm to charmhub
+        uses: canonical/charming-actions/upload-charm@2.2.2
+        with:
+          credentials: "${{ secrets.CHARMHUB_TOKEN }}"
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+          channel: "6/edge"
```
Comment (MiaAltieri) [inline .github/workflows/release.yaml:39]: how exciting!!

**Review by MiaAltieri — DISMISSED** [review]
Thank you for all your work Pedro

File: tests/integration/relation_tests/legacy_relations/helpers.py [hunk 2]
```diff
@@ -136,7 +136,7 @@ async def mongo_tls_command(ops_test: OpsTest) -> str:
     replica_set_uri = f"mongodb://{hosts}/admin?replicaSet={app}"
 
     return (
-        f"charmed-mongodb.mongo '{replica_set_uri}'  --eval 'rs.status()'"
+        f"charmed-mongodb.mongosh '{replica_set_uri}'  --eval 'rs.status()'"
```
Comment (dmitry-ratushnyy) [inline tests/integration/relation_tests/legacy_relations/helpers.py:139]: I see we use this `charmed-mongodb.mongosh`  in many places, this looks like a good candidate to be extracted to a constant. 

Maybe we can do that as a part of this PR? What do you think?

Comment (delgod) [reply tests/integration/relation_tests/legacy_relations/helpers.py:139]: yes, please
```suggestion
        f"{MONGO_SHELL} '{replica_set_uri}'  --eval 'rs.status()'"
```

**Review by delgod — APPROVED** [review]

**Review by dmitry-ratushnyy — APPROVED** [review]

**Review by MiaAltieri — APPROVED** [review]

**Pull request merged**